packages/shopify-app-remix/src/server/types.ts

```diff
@@ -101,6 +97,166 @@ type SessionStorageType<Config extends AppConfigArg> =
+   *
+   * const shopify = shopifyApp({
+   *   restResources,
+   *   // ...etc
+   * });
+   * export default shopify;
+   * export const authenticate = shopify.authenticate;
+   *
+   * // app/routes/**\/*.jsx
+   * import { LoaderArgs, json } from "@remix-run/node";
+   * import { authenticate } from "../../shopify.server";
+   *
+   * export async function loader({ request }: LoaderArgs) {
+   *   const {admin, session, sessionToken, billing} = authenticate.admin(request);
+   *
+   *   return json(await admin.rest.resources.Product.count({ session }));
+   * }
+   * ```
+   */
+  admin: AuthenticateAdmin<Config, RestResourcesType<Config>>;
+
+  /**
+   * Authenticate a public request and get back a session token
+   *
+   * An example of when to use this is a request from a checkout extension.
+   *
+   * @example
+   * Authenticating a request from a checkout extension
+   *
+   * ```ts
+   * // app/routes/api/checkout.jsx
+   * import { LoaderArgs, json } from "@remix-run/node";
+   * import { authenticate } from "../../shopify.server";
+   * import { getWidgets } from "~/db/widgets";
+   *
+   * export async function loader({ request }: LoaderArgs) {
+   *   const {sessionToken} = authenticate.public(request);
+   *
+   *   return json(await getWidgets(sessionToken));
+   * }
+   * ```
+   */
+  public: AuthenticatePublic;
+
+  /**
+   * Authenticate a Shopify webhook request, get back an authenticated admin context and details on the webhook request
+   *
+   * @example
+   * Authenticating a webhook request
+   *
+   * ```ts
+   * // app/routes/api/checkout.jsx
+   * import {
+   *   DeliveryMethod,
+   *   shopifyApp,
+   * } from "@shopify/shopify-app-remix";
+   *
+   * const shopify = shopifyApp({
+   *   webhooks: {
+   *    APP_UNINSTALLED: {
+   *       deliveryMethod: DeliveryMethod.Http,
+   *       callbackUrl: "/webhooks",
+   *     },
+   *   },
+   *   hooks: {
+   *     afterAuth: async ({ session }) => {
+   *       shopify.registerWebhooks({ session });
+   *     },
+   *   },
+   *   // ...etc
+   * });
+   * export default shopify;
+   * export const authenticate = shopify.authenticate;
+   *
+   * // app/routes/webhooks.ts
+   * import { ActionArgs } from "@remix-run/node";
+   * import { authenticate } from "../shopify.server";
+   * import db from "../db.server";
+   *
+   * export const action = async ({ request }: ActionArgs) => {
+   *   const { topic, shop, session } = await authenticate.webhook(request);
+   *
+   *   switch (topic) {
+   *     case "APP_UNINSTALLED":
+   *       if (session) {
+   *         await db.session.deleteMany({ where: { shop } });
+   *       }
+   *       break;
+   *     case "CUSTOMERS_DATA_REQUEST":
+   *     case "CUSTOMERS_REDACT":
+   *     case "SHOP_REDACT":
+   *     default:
+   *       throw new Response("Unhandled webhook topic", { status: 404 });
+   *   }
+   *
+   *   throw new Response();
+   * };
+   * ```
+   */
```

Comment:
Anyway we can pull all these examples into the docs?  Theres a lot of detail in the intellisense docs that isn't in the shopify.dev docs.